tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/SDKBuildPropertiesResourceListener.java

```diff
@@ -0,0 +1,215 @@
+import org.eclipse.core.resources.IResourceDeltaVisitor;
+import org.eclipse.core.resources.WorkspaceJob;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+
+/**
+ * @author Simon Jiang
+ */
+public class SDKBuildPropertiesResourceListener implements IResourceChangeListener, IResourceDeltaVisitor
+{
+
+    private static final String ID_SDK_PROPERTIES_INVALID = "sdk-properties-invalid";
+
+
+    private final static Pattern PATTERN_BUILD_PROPERTIES  = Pattern.compile("build.[\\w|\\W.]*properties");
+
+    protected void processPropertiesFileChanged() throws CoreException
+    {
+
+        final SDK sdk = SDKUtil.getWorkspaceSDK();
+
+        if ( sdk != null )
+        {
+            final IProject sdkProject = SDKUtil.getWorkspaceSDKProject();
+
+            MarkerUtil.clearMarkers(sdkProject, IMarker.PROBLEM, ID_SDK_PROPERTIES_INVALID);
+
+            final IStatus sdkValid = sdk.validate( true );
+
+            if ( sdkValid.isOK() )
+            {
+                for( final IProject project : CoreUtil.getAllProjects() )
+                {
+                    if ( SDKUtil.isSDKProject( project ) && sdk.getLocation().isPrefixOf( project.getLocation() ))
+                    {
+                        Job job = new WorkspaceJob( "Updating project setting base on new sdk properties for " + project.getName() )
+                        {
+                            @Override
+                            public IStatus runInWorkspace( final IProgressMonitor monitor ) throws CoreException
+                            {
+                                ClasspathUtil.updateRequestContainer(project);
+                                return Status.OK_STATUS;
+                            }
+                        };
+                        job.setRule( project );
+                        job.schedule();
+                    }
+                }
+            }
+            else
+            {
+                final IStatus[] sdkValids = sdkValid.getChildren();
+
+                for( final IStatus iStatus : sdkValids )
+                {
+                    MarkerUtil.setMarker(sdkProject, IMarker.PROBLEM, IMarker.SEVERITY_ERROR,
+                        iStatus.getMessage(), sdkProject.getFullPath().toPortableString(),ID_SDK_PROPERTIES_INVALID);
+                }
+
+                return;
+            }
+        }
+    }
+
+    @Override
+    public void resourceChanged( final IResourceChangeEvent event )
+    {
+        if( event == null )
+        {
+            return;
+        }
+
+        try
+        {
+            final SDK sdk = SDKUtil.getWorkspaceSDK();
+
+            if ( sdk != null )
+            {
+                String[] customerPropertyNames = sdk.getBuildPropertyFileNames();
```

Comment:
We don't need to do this call, this will be too slow to do for every change.  We should instead just find the path of each delta and match the latest segment against our build.properties pattern above instead.